How KIC Works section (#1080)
Kong/developer.konghq.com: app/kubernetes-ingress-controller/reference/annotations.md

| REQUIRED [`kubernetes.io/ingress.class`](#kubernetesioingressclass) | Restrict the KongConsumers that a controller should satisfy            |
| [`konghq.com/plugins`](#konghq-com-plugins)                         | Run plugins for a specific consumer                                    |
| [`konghq.com/tags`](#konghq-com-tags)                               | Assign custom tags to Kong entities generated out of this KongConsumer |

`kubernetes.io/ingress.class` is normally required, and its value should match the value of the `--ingress-class` controller argument (`kong` by default).

Setting the `--process-classless-kong-consumer` controller flag removes that requirement. when it is enabled, the controller processes KongConsumers with no `kubernetes.io/ingress.class` annotation. Recommended best practice is to set the annotation and leave this flag disabled; the flag is primarily intended for older configurations, as controller versions prior to 0.10 processed classless KongConsumer resources by default.

## Service annotations

These annotations are supported on Service resources.

> | Annotation name                                                                   | Description                                                                                                                            |

Reviewer: Replace this table with:

```
{% table %}
columns:
  - title: Annotation name
    key: name
  - title: Description
    key: description
rows:
  - name: "[`konghq.com/plugins`](#konghq-com-plugins)"
    description: "Run plugins for a specific Service"
  - name: "[`konghq.com/protocol`](#konghq-com-protocol)"
    description: "Set protocol {{site.base_gateway}} should use to talk to a Kubernetes service"
  - name: "[`konghq.com/path`](#konghq-com-path)"
    description: "HTTP Path that is always prepended to each request that is forwarded to a Kubernetes service"
  - name: "[`konghq.com/client-cert`](#konghq-com-client-cert)"
    description: "Client certificate and key pair {{site.base_gateway}} should use to authenticate itself to a specific Kubernetes service"
  - name: "[`konghq.com/host-header`](#konghq-com-host-header)"
    description: "Set the value sent in the `Host` header when proxying requests upstream"
  - name: "[`ingress.kubernetes.io/service-upstream`](#ingresskubernetesioservice-upstream)"
    description: "Offload load-balancing to kube-proxy or sidecar"
  - name: "[`konghq.com/upstream-policy`](#konghq-com-upstream-policy)"
    description: "Override {{site.base_gateway}} Upstream configuration with KongUpstreamPolicy resource"
  - name: "[`konghq.com/connect-timeout`](#konghq-com-connecttimeout)"
    description: "Set the timeout for completing a TCP connection"
  - name: "[`konghq.com/read-timeout`](#konghq-com-readtimeout)"
    description: "Set the timeout for receiving an HTTP response after sending a request"
  - name: "[`konghq.com/write-timeout`](#konghq-com-writetimeout)"
    description: "Set the timeout for writing data"
  - name: "[`konghq.com/retries`](#konghq-com-retries)"
    description: "Set the number of times to retry requests that failed"
  - name: "[`konghq.com/tags`](#konghq-com-tags)"
    description: "Assign custom tags to {{site.base_gateway}} entities generated out of this Service"
  - name: "[`konghq.com/tls-verify`](#konghq-com-tls-verify)"
    description: "Enable or disable verification of the upstream service's TLS certificates"
  - name: "[`konghq.com/tls-verify-depth`](#konghq-com-tls-verify-depth)"
    description: "Set the maximal depth of a certificate chain when verifying the upstream service's TLS certificates"
  - name: "[`konghq.com/ca-certificates-secrets`](#konghq-com-ca-certificates-secrets)"
    description: "Assign CA certificates Secrets to be used for the upstream service's TLS certificates verification"
  - name: "[`konghq.com/ca-certificates-configmaps`](#konghq-com-ca-certificates-configmaps)"
    description: "Assign CA certificates ConfigMaps to be used for the upstream service's TLS certificates verification"
{% endtable %}
```

| [`konghq.com/tags`](#konghq-com-tags)                                             | Assign custom tags to Kong entities generated out of this Service                                                                      |
| [`konghq.com/tls-verify`](#konghq-com-tls-verify)                                 | Enable or disable verification of the upstream service's TLS certificates                                                              |
| [`konghq.com/tls-verify-depth`](#konghq-com-tls-verify-depth)                     | Set the maximal depth of a certificate chain when verifying the upstream service's TLS certificates                                    |
| [`konghq.com/ca-certificates-secrets`](#konghq-com-ca-certificates-secrets)       | Assign CA certificates Secrets to be used for the upstream service's TLS certificates verification                                     |
| [`konghq.com/ca-certificates-configmaps`](#konghq-com-ca-certificates-configmaps) | Assign CA certificates ConfigMaps to be used for the upstream service's TLS certificates verification                                  |

### konghq.com/protocol

> This annotation can be set on a Kubernetes Service resource and indicates the protocol that should be used by Kong to communicate with the service. In other words, the protocol is used for communication between a [Kong Service](/gateway/entities/service/) and a Kubernetes Service, internally in the Kubernetes cluster.

Reviewer: ```suggestion
This annotation can be set on a Kubernetes Service resource and indicates the protocol that should be used by {{site.base_gateway}} to communicate with the Service. In other words, the protocol is used for communication between a [Kong Service](/gateway/entities/service/) and a Kubernetes Service, internally in the Kubernetes cluster.
```

Accepted values are:

- `http`
- `https`
- `grpc`
- `grpcs`
- `tcp`
- `tls`

### konghq.com/path

> This annotation can be used on a Service resource only.  This annotation can be used to prepend an HTTP path of a request, before the request is forwarded.

Reviewer: ```suggestion
This annotation can be used on a Service resource only and can be used to prepend an HTTP path of a request before the request is forwarded.
```

```

### ingress.kubernetes.io/service-upstream

By default, the {{site.kic_product_name}} distributes traffic amongst all the Pods of a Kubernetes Service by forwarding the requests directly to Pod IP addresses. One can choose the load-balancing strategy to use by specifying a KongIngress resource.

However, in some use-cases, the load-balancing should be left up to `kube-proxy`, or a sidecar component in the case of Service Mesh deployments.

Setting this annotation to a Service resource in Kubernetes will configure the {{site.kic_product_name}} to directly forward the traffic outbound for this Service to the IP address of the service (usually the ClusterIP).

`kube-proxy` can then decide how it wants to handle the request and route the traffic accordingly. If a sidecar intercepts the traffic from the controller, it can also route traffic as it sees fit in this case.

Following is an example snippet you can use to configure this annotation on a Service resource in Kubernetes, (please note the quotes around `true`):

```yaml
annotations:
  ingress.kubernetes.io/service-upstream: "true"
```

### konghq.com/upstream-policy {% new_in 3.0 %}

> This annotation can be used to attach `KongUpstreamPolicy` resources to `Services`. The value of the annotation is the name of the `KongUpstreamPolicy` object in the same namespace as the `Service`. Please refer to the [KongUpstreamPolicy reference](/kubernetes-ingress-controller/reference/custom-resources/#kongupstreampolicy) for details on how to configure the `KongUpstreamPolicy` resource.

Reviewer: ```suggestion
This annotation can be used to attach `KongUpstreamPolicy` resources to `Services`. The value of the annotation is the name of the `KongUpstreamPolicy` object in the same namespace as the `Service`. See the [KongUpstreamPolicy reference](/kubernetes-ingress-controller/reference/custom-resources/#kongupstreampolicy) for details on how to configure the `KongUpstreamPolicy` resource.
```
